src/app/home-page/widget/entry-box/entry-box.component.ts

```diff
@@ -77,29 +79,27 @@ export class EntryBoxComponent extends Base implements OnInit {
   }
 
   getMyEntries() {
-    this.usersService
-      .getUserEntries(null, this.filterText)
+    if (this.entryType === EntryUpdateTime.EntryTypeEnum.TOOL) {
+      this.userEntries$ = this.usersService.getUserEntries(null, this.filterText, 'TOOLS');
+    } else if (this.entryType === EntryUpdateTime.EntryTypeEnum.WORKFLOW) {
+      this.userEntries$ = this.usersService.getUserEntries(null, this.filterText, 'WORKFLOWS');
+    } else {
+      this.userEntries$ = this.usersService.getUserEntries(null, this.filterText, 'SERVICES');
+    }
+    this.userEntries$
       .pipe(
         finalize(() => (this.isLoading = false)),
         debounceTime(750),
         takeUntil(this.ngUnsubscribe)
       )
       .subscribe(
         (myEntries: Array<EntryUpdateTime>) => {
-          this.listOfEntries = [];
-          this.totalEntries = 0;
-          myEntries
-            .filter(
-              (entry) =>
-                entry.entryType === this.entryType ||
-                (this.entryType === EntryUpdateTime.EntryTypeEnum.TOOL && entry.entryType === EntryUpdateTime.EntryTypeEnum.APPTOOL)
-            )
-            .forEach((entry: EntryUpdateTime) => {
-              if (this.listOfEntries.length < 7) {
-                this.listOfEntries.push(entry);
-              }
-              this.totalEntries += 1;
-            });
+          myEntries.forEach((entry: EntryUpdateTime) => {
```

Comment:
It was already like this, but you could do, also untested:

```
this.listOfEntries = myEntries.slice(0,7); // Will work even if there are less than 7 entries
this.totalEntries = myEntries.length;
```

More readable, IMHO, although your opinion may vary. :)